crates/oxc_prettier/src/printer/mod.rs

```diff
@@ -211,6 +214,37 @@ impl<'a> Printer<'a> {
             if remaining_width < 0 {
                 return false;
             }
+
+            if check_cmds && queue.is_empty() {
```

Comment:
What about this block? I can't find the relevant code in prettier 😅

Reply:
I didn't look closely at prettier's code, which was written to solve this case

`* import/inline.js`

Reply:
I think it's best to port most of prettier's code, otherwise it is going to be really hard to debug things :-(

Reply:
The prettier's doc printer is based on a paper, I think it is best to port line by line 🤔 See https://prettier.io/docs/en/technical-details

Reply:
Okay, I'll dive into prettier

Reply:
I noticed that the original implementation doesn't match the prettier's implementation. This block aligns with prettier, but still needs to improve

Reply:
> original implementation
Which implementation?

What I hope to see is an alignment with https://github.com/prettier/prettier/blob/e3fe2a1f012e5426e007510743454f436cbba9df/src/document/printer.js#L196-L300

Reply:
I refactored it and now the implementation is similar to prettier. The regression test seems to need to be implemented in a different way